libs/polysemy-wire-zoo/src/Wire/Sem/Logger/TinyLog.hs

```diff
@@ -0,0 +1,63 @@
+-- This file is part of the Wire Server implementation.
+--
+-- Copyright (C) 2022 Wire Swiss GmbH <opensource@wire.com>
+--
+-- This program is free software: you can redistribute it and/or modify it under
+-- the terms of the GNU Affero General Public License as published by the Free
+-- Software Foundation, either version 3 of the License, or (at your option) any
+-- later version.
+--
+-- This program is distributed in the hope that it will be useful, but WITHOUT
+-- ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+-- FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+-- details.
+--
+-- You should have received a copy of the GNU Affero General Public License along
+-- with this program. If not, see <https://www.gnu.org/licenses/>.
+
+module Wire.Sem.Logger.TinyLog
+  ( loggerToTinyLog,
+    loggerToTinyLogReqId,
+    stringLoggerToTinyLog,
+    discardLogs,
+    discardTinyLogs,
+    module Wire.Sem.Logger.Level,
+  )
+where
+
+import Data.Id
+import Imports
+import Polysemy
+import qualified System.Logger as Log
+import Wire.Sem.Logger
+import Wire.Sem.Logger.Level
+
+loggerToTinyLog ::
+  Member (Embed IO) r =>
+  Log.Logger ->
+  Sem (Logger (Log.Msg -> Log.Msg) ': r) a ->
+  Sem r a
+loggerToTinyLog tinylog = interpret $ \case
+  Log lvl msg ->
+    embed @IO $ Log.log tinylog (toLevel lvl) msg
+
+loggerToTinyLogReqId ::
+  Member (Embed IO) r =>
+  Log.Logger ->
+  RequestId ->
+  Sem (Logger (Log.Msg -> Log.Msg) ': r) a ->
+  Sem r a
+loggerToTinyLogReqId tinylog r = interpret $ \case
+  Log lvl msg ->
+    embed @IO
+      . Log.log tinylog (toLevel lvl)
+      $ Log.field "request" (unRequestId r) Log.~~ msg
+
+stringLoggerToTinyLog :: Member (Logger (Log.Msg -> Log.Msg)) r => Sem (Logger String ': r) a -> Sem r a
+stringLoggerToTinyLog = mapLogger @String Log.msg
+
+discardLogs :: Sem (Logger msg ': r) a -> Sem r a
+discardLogs = interpret $ \(Log _ _) -> pure ()
+
+discardTinyLogs :: Sem (Logger (Log.Msg -> Log.Msg) ': r) a -> Sem r a
```

Comment:
Why do you need this function?

Reply:
This is a help in type inference. If I don't use it, then GHC reports overlapping instances. An alternative is to use `discardLogs @(Log.Msg -> Log.Msg)` instead of `discardTinyLogs`.

Reply:
That's surprising to me. Does the polysemy plugin help?

Reply:
> That's surprising to me. Does the polysemy plugin help?

It doesn't seem to. I tried this: 1) put the plugin as a GHC option in the cabal file and make its package a dependency, 2) use `discardLogs` instead of `discardTinyLogs` where I'm hoping the plugin would help, 3) observe I get the aforementioned error on overlapping instances, e.g.:

```
src/Federator/MockServer.hs:93:15: error:
    * Overlapping instances for Member
                                  (Wire.Sem.Logger.Logger
                                     (System.Logger.Message.Msg -> System.Logger.Message.Msg))
                                  '[Wire.Sem.Logger.Logger msg0, Embed IO]
        arising from a use of `runWaiErrors'
      Matching instances:
        two instances involving out-of-scope types
        (use -fprint-potential-instances to see them all)
      (The choice depends on the instantiation of `msg0'
       To pick the first instance above, use IncoherentInstances
       when compiling the other instance declarations)
    * In the second argument of `(.)', namely
        `runWaiErrors @'[ValidationError, ServerError, MockException]'
      In the second argument of `(.)', namely
        `discardLogs
           . runWaiErrors @'[ValidationError, ServerError, MockException]'
      In the expression:
        runM
          . discardLogs
              . runWaiErrors @'[ValidationError, ServerError, MockException]
   |
93 |             . runWaiErrors
   |               ^^^^^^^^^^^^...
cabal: Failed to build federator-1.0.0 (which is required by brig-2.0).
```